ScrubBot.Modules/EventModule.cs

```diff
@@ -0,0 +1,126 @@
+﻿using Discord;
+using Discord.Commands;
+using Discord.WebSocket;
+
+using Microsoft.EntityFrameworkCore.Internal;
+
+using ScrubBot.Domain;
+using ScrubBot.Extensions;
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace ScrubBot.Modules
+{
+    public class EventModule : Module
+    {
+        public EventModule()
+        {
+
+        }
+
+        [Command("ShowEvents"), Summary("Show the first 10 upcoming events")]
+        public async Task ShowEvents(int amount = 10)
+        {
+            List<Event> events = Database.Events
+                .Where(x => x.Guild.Id == Context.Guild.Id)
+                .OrderByDescending(x => x.CreationDate)
+                .Take(amount)
+                .ToList();
+
+            if (events.Count is 0)
+            {
+                await ReplyAsync(new EmbedBuilder().CreateError("There are no events to be displayed!"));
+                return;
+            }
+
+            await ReplyAsync(new EmbedBuilder().CreateEventEmbed("Upcoming events", events));
+        }
+
+        // @ScrubBot CreateEvent "Title goes here" "Description goes here" 24/10/2018 20
+        [Command("CreateEvent"), Summary("Create a new event")]
+        public async Task CreateEvent(string eventTitle, string description, DateTime occurenceDateTimeUTC, int maxSubscribers)
+        {
+            if (Database.Events.Any(x => x.Title == eventTitle && x.Guild.Id == Context.Guild.Id))
+            {
+                await ReplyAsync(new EmbedBuilder().CreateError($"An event with the name **{eventTitle}** already exists!"));
+                return;
+            }
+
+            Event newEvent = new Event
+            {
+                Title = eventTitle,
+                Description = description,
+                Guild = base.Guild, 
+                OccurenceDate = occurenceDateTimeUTC.ToUniversalTime(),
+                Author = base.User,
+                MaxSubscribers = maxSubscribers
+            };
+
+            await Database.Events.AddAsync(newEvent);
+
+            await ReplyAsync(new EmbedBuilder().CreateSuccess($"Event **{eventTitle}** has been successfully created for **{occurenceDateTimeUTC}** with a max of **{maxSubscribers}** subscribers!"));
+        }
+
+        [Command("JoinEvent"), Summary("Join a specific event")]
+        public async Task JoinEvent(string eventTitle)
+        {
+            bool hasEvent = Database.Events.Any(x => x.Title == eventTitle);
+            bool isSameServer = Database.Events.Any(x => x.Guild.Id == Context.Guild.Id);
```

Comment:
```suggestion
            bool isSameServer = Database.Events.Any(x => x.Guild.Id == Guild.Id);
```

Reply:
Module base class already has access to the Guild and User models from the database.